packages/oslo-converter-uml-ea/lib/converter-handlers/ConnectorConverterHandler.ts

```diff
@@ -1,153 +1,233 @@
 import type { QuadStore } from '@oslo-flanders/core';
 import { ns } from '@oslo-flanders/core';
-import type { DataRegistry, EaConnector, EaElement, EaTag } from '@oslo-flanders/ea-uml-extractor';
-import { ConnectorType, NormalizedConnector, NormalizedConnectorTypes } from '@oslo-flanders/ea-uml-extractor';
+import type { DataRegistry } from '@oslo-flanders/ea-uml-extractor';
+import {
+  ConnectorType,
+  NormalizedConnector,
+  NormalizedConnectorTypes,
+} from '@oslo-flanders/ea-uml-extractor';
 import type * as RDF from '@rdfjs/types';
-import { injectable } from 'inversify';
+import { inject, injectable } from 'inversify';
 import { TagNames } from '../enums/TagNames';
 import { ConverterHandler } from '../interfaces/ConverterHandler';
 import type { UriRegistry } from '../UriRegistry';
-import { convertToCase, getTagValue, ignore } from '../utils/utils';
+import { getTagValue, ignore } from '../utils/utils';
+import { ConnectorNormalisationService } from '../ConnectorNormalisationService';
+import { EaUmlConverterServiceIdentifier } from '../config/EaUmlConverterServiceIdentifier';
 
 @injectable()
 export class ConnectorConverterHandler extends ConverterHandler<NormalizedConnector> {
-  public async filterIgnoredObjects(model: DataRegistry): Promise<DataRegistry> {
-    model.connectors = model.connectors.filter(x => !ignore(x));
+  @inject(EaUmlConverterServiceIdentifier.ConnectorNormalisationService)
+  private readonly connectorNormalisationService!: ConnectorNormalisationService;
+
+  public async filterIgnoredObjects(
+    model: DataRegistry
+  ): Promise<DataRegistry> {
+    model.connectors = model.connectors.filter((x) => !ignore(x));
 
     return model;
   }
 
-  public async convert(model: DataRegistry, uriRegistry: UriRegistry, store: QuadStore): Promise<QuadStore> {
+  public async convert(
+    model: DataRegistry,
+    uriRegistry: UriRegistry,
+    store: QuadStore
+  ): Promise<QuadStore> {
     model.normalizedConnectors
-      .filter(x => model.targetDiagram.connectorsIds.includes(x.originalId))
-      .forEach(object => store.addQuads(this.createQuads(object, uriRegistry, model)));
+      .filter((x) => model.targetDiagram.connectorsIds.includes(x.originalId))
+      .forEach((object) =>
+        store.addQuads(this.createQuads(object, uriRegistry, model))
+      );
 
     return store;
   }
 
   public async normalize(model: DataRegistry): Promise<DataRegistry> {
-    model.normalizedConnectors = model.connectors
-      .filter(x => model.targetDiagram.connectorsIds.includes(x.id))
-      .flatMap(x => this.normalizeConnector(x, model.elements));
+    const tasks: Promise<NormalizedConnector[]>[] = [];
```

Comment:
This can be converted into a `.map` function

```js
    const tasks: Promise<NormalizedConnector[]>[] = model.connectors.map(connector =>
      this.connectorNormalisationService.normalise(connector, model));
```